wagtail/migrations/0086_populate_grouppagepermission_permission.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 4.2.1 on 2023-06-14 16:23
+
+from django.contrib.auth.management import create_permissions
+from django.db import migrations, models
+from django.db.models.functions import Concat
+
+
+def add_permissions(apps, schema_editor):
+    # Ensure all the new permissions in Page.Meta.permissions are created
+    app_config = apps.get_app_config("wagtailcore")
+    app_config.models_module = True
+    create_permissions(app_config, verbosity=0)
+    app_config.models_module = None
+
+
+def populate_grouppagepermission_permission(apps, schema_editor):
+    ContentType = apps.get_model("contenttypes.ContentType")
+    Permission = apps.get_model("auth.Permission")
+    GroupPagePermission = apps.get_model("wagtailcore.GroupPagePermission")
+
+    # Change it to match Django's permission codename
+    GroupPagePermission.objects.filter(permission_type="edit").update(
+        permission_type="change"
+    )
+
+    page_type = ContentType.objects.get(app_label="wagtailcore", model="page")
+    GroupPagePermission.objects.all().update(
+        permission=Permission.objects.filter(
+            content_type=page_type,
+            codename=Concat(models.OuterRef("permission_type"), models.Value("_page")),
+        ).values_list("pk", flat=True)[:1]
+    )
+
+
+def empty_grouppagepermission_permission(apps, schema_editor):
+    GroupPagePermission = apps.get_model("wagtailcore.GroupPagePermission")
+    GroupPagePermission.objects.all().update(permission=None)
```

Comment:
Unfortunately that's not enough, because undoing the final migration (0087) currently isn't even possible, as that means we'll be undoing `RemoveField` on `permission_type` which becomes an `AddField`, that fails because the `permission_type` is not null.

Looks like we'll need to add a migration that makes it nullable first...